File: regexlib/2021-5-15/javascript_test_file/regexlib_8548.js
/* 8548
 * (?:\b\w*(\w\w?)\1{2,}\w*\b)
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"111"+"00"*256+"! _1_EOA(iii)"
 */
var REGEX = /(?:\b\w*(\w\w?)\1{2,}\w*\b)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '111' + '00'.repeat(i*1) + '! _1_EOA(iii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_8548.py
# 8548
# (?:\b\w*(\w\w?)\1{2,}\w*\b)
# EXPONENT
# nums:5
# EXPONENT AttackString:"111"+"00"*512+"! _1_EOA(iii)"

import re2 as re
from time import perf_counter

regex = """(?:\b\w*(\w\w?)\1{2,}\w*\b)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "111" + "00" * i * 1 + "! _1_EOA(iii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")